Media Service Data Progress Bar Improvements (#749)
smartdevicelink/sdl_evolution: proposals/xxx-media-service-data-progress-bar-improvements.md

# Add Currently Playing Media Image to MediaServiceData

* Proposal: [SDL-xxx](xxx-media-service-data-progress-bar-improvements.md)
* Author: [Jack Byrne](https://github.com/jacklivio)
* Status: **Awaiting review**
* Impacted Platforms: [Core / iOS / Java Suite / RPC]

## Introduction

> This proposal adds new paramters to the media service data to allow the app consumers know the playback status and progress of a selected track.

Reviewer: ```suggestion
This proposal adds new parameters to the media service data to allow the app consumers to know the playback status and progress of a selected track.
```

## Motivation

> The movitation for the proposal came from creating an implementation of a media type app service consumer that displays the progress bar data from an active media service provider.

Reviewer: ```suggestion
The motivation for the proposal came from creating an implementation of a media type app service consumer that displays the progress bar data from an active media service provider.
```

> Currently app service providers can send notify consumers about a tracks playback progress via `trackPlaybackProgress` and `trackPlaybackDuration`. However, if an app consumer wants to display an updating progress using these parameters, it would require the app provider to send an OnAppServiceData RPC with updated information every second.

Reviewer: ```suggestion
Currently app service providers can notify consumers about a track's playback progress via `trackPlaybackProgress` and `trackPlaybackDuration`. However, if an app consumer wants to display progress using these parameters, it would require the app provider to send an `OnAppServiceData` RPC with updated information every second.
```

> Also, there are no paramters in the media service data struct that tell the consumer if the track is playing, paused, or stopped. 

Reviewer: ```suggestion
Also, there are no parameters in the media service data struct that tell the consumer if the track is playing, paused, or stopped. 
```
